### PR TITLE
[MXNET-641] fix R windows install docs

### LINKED ISSUE #8936: Issue on MXNet R installation with GPU support on Windows
## Description
When following the installation tutorial : https://gist.github.com/thirdwing/89aa9bfc588ade138496e6932072152c, the step : **R CMD INSTALL --no-multiarch R-package** returns 
**"Error: compilation failed for package 'mxnet'"** ( several errors including : compiler and library support for the ISO C++ 2011 standard, im2rec.cc:20:30: fatal error, make: *** [im2rec.o] Error 1,...)

## Environment info (Required)
Windows 10
GPU : GeForce GTX 1080 Ti
R 3.3.2
Rtools 3.4
Visual Studio 14
Cuda 9
Cudnn 7
OpenCV 3.3.1
prebuilt GPU-enabled MXNet libraries : 
  -> 20171204_mxnet_x64_vc141_gpu_cu90
  -> prebuildbase_win10_x64_vc14_v2

All environment variables seem to be well defined. 
For Rtools, paths are : C:\Rtools\mingw_64\bin & C:\Rtools\bin

## R sessionInfo:
R version 3.3.2 (2016-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

locale:
[1] LC_COLLATE=French_Switzerland.1252  LC_CTYPE=French_Switzerland.1252    LC_MONETARY=French_Switzerland.1252
[4] LC_NUMERIC=C                        LC_TIME=French_Switzerland.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] tools_3.3.2

## Error Message:
c:\mxnet>R CMD INSTALL --no-multiarch R-package
* installing to library 'C:/R-3.3.2/library'
* installing *source* package 'mxnet' ...
** libs
c:/Rtools/mingw_64/bin/g++  -I"C:/R-3.3.2/include" -DNDEBUG -I../inst/include   -I"C:/R-3.3.2/library/Rcpp/include" -I"C:/TDM-GCC-64//include"     -O2 -Wall  -mtune=core2 -c executor.cc -o executor.o
In file included from executor.cc:9:0:
./base.h: In function 'std::string mxnet::R::toPyString(const string&, const RObject&)':
./base.h:276:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < vec.size(); ++i) {
                          ^
./base.h: In function 'std::vector<unsigned int> mxnet::R::Dim2InternalShape(const Rcpp::Dimension&)':
./base.h:340:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < rshape.size(); ++i) {
                        ^
executor.cc: In static member function 'static void mxnet::R::Executor::UpdateArray(const char*, const List&, Rcpp::List*, bool, bool)':
executor.cc:43:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < from.size(); ++i) {
                          ^
executor.cc: In static member function 'static Rcpp::List mxnet::R::Executor::CloneArray(const List&)':
executor.cc:87:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < src.size(); ++i) {
                        ^
executor.cc: In function 'Rcpp::List* mxnet::R::CreateArrayList(const List&, const string&, const std::vector<std::basic_string<char> >&, const RObjectType&, std::vector<void*>*)':
executor.cc:123:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < source_array.size(); ++i) {
                          ^
executor.cc: In function 'Rcpp::List* mxnet::R::CreateGradList(const List&, const List&, const std::vector<std::basic_string<char> >&, const RObjectType&, std::vector<void*>*, std::vector<unsigned int>*)':
executor.cc:155:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < grad_reqs.size(); ++i) {
                          ^
c:/Rtools/mingw_64/bin/g++  -I"C:/R-3.3.2/include" -DNDEBUG -I../inst/include   -I"C:/R-3.3.2/library/Rcpp/include" -I"C:/TDM-GCC-64//include"     -O2 -Wall  -mtune=core2 -c export.cc -o export.o
In file included from export.cc:9:0:
./base.h: In function 'std::string mxnet::R::toPyString(const string&, const RObject&)':
./base.h:276:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < vec.size(); ++i) {
                          ^
./base.h: In function 'std::vector<unsigned int> mxnet::R::Dim2InternalShape(const Rcpp::Dimension&)':
./base.h:340:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < rshape.size(); ++i) {
                        ^
export.cc: In static member function 'static void mxnet::R::Exporter::Export(const string&)':
export.cc:92:24: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for (size_t i = 0; i < func_names.size(); ++i) {
                        ^
c:/Rtools/mingw_64/bin/g++  -I"C:/R-3.3.2/include" -DNDEBUG -I../inst/include   -I"C:/R-3.3.2/library/Rcpp/include" -I"C:/TDM-GCC-64//include"     -O2 -Wall  -mtune=core2 -c im2rec.cc -o im2rec.o
In file included from C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/random:35:0,
                 from im2rec.cc:14:
**C:/Rtools/mingw_64/x86_64-w64-mingw32/include/c++/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^**
****In file included from ../inst/include/dmlc/io.h:14:0,
                 from im2rec.cc:16:
../inst/include/dmlc/./logging.h:67:3: warning: identifier 'nullptr' is a keyword in C++11 [-Wc++0x-compat]
   LogCheckError() : str(nullptr) {}
   ^**
im2rec.cc:20:30: fatal error: opencv2/opencv.hpp: No such file or directory
 #include <opencv2/opencv.hpp>**
                              ^
compilation terminated.
**make: *** [im2rec.o] Error 1
Warning: l'exécution de la commande 'make -f "Makevars.win" -f "C:/R-3.3.2/etc/x64/Makeconf" -f "C:/R-3.3.2/share/make/winshlib.mk" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="mxnet.dll" WIN=64 TCLBIN=64 OBJECTS="executor.o export.o im2rec.o io.o kvstore.o mxnet.o ndarray.o symbol.o"' renvoie un statut 2
ERROR: compilation failed for package 'mxnet'**
* removing 'C:/R-3.3.2/library/mxnet'

### PR DESCRIPTION
**Pull request by ankkhedia**
## Description ##
This PR fixes the install instructions for R in Windows. It will fix 
#8936 #8927 #9952 

### Essentials ###
Please feel free to remove inapplicable items for your PR.

### Changes ###
It fixed documentation for MXNet-R installation on Windows
## Comments ##
The changes are complete , however the binaries are yet to be put in the bucket for GPU prebuild binaries instructions to work

### CODE REVIEW

File: docs/install/windows_setup.md [hunk 1]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
-7. Copy the following shared libraries (.dll files) into the ```R-package/inst/libs/x64``` folder:
-```
-cublas64_80.dll
-cudart64_80.dll
-cudnn64_5.dll
-curand64_80.dll
+2. Download prebuilt GPU-enabled MXNet libraries for Windows from https://github.com/yajiedesign/mxnet/releases. You will need `mxnet_x64_vc14_gpu_cuX.7z` and `prebuildbase_win10_x64_vc14.7z` where X stands for your CUDA toolkit version
+3. Create a folder called ```R-package/inst/libs/x64```. MXNet supports only 64-bit operating systems, so you need the x64 folder.
+4. Copy the following shared libraries (.dll files) into the ```R-package/inst/libs/x64``` folder:
+```
```
Comment (anirudhacharya) [inline docs/install/windows_setup.md]: why were these removed -
-cublas64_80.dll	
-cudart64_80.dll	
-cudnn64_5.dll	
-curand64_80.dll

Comment (ankkhedia) [reply docs/install/windows_setup.md]: These don't need to be there in the R package libraries,it is taken from cuda toolkit installation and not here

File: docs/install/windows_setup.md [hunk 2]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
+#### Building MXNet from Source Code(CPU)
+1. Clone the MXNet github repo.
 
-```r
-  cran <- getOption("repos")
-  cran["dmlc"] <- "https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/R/CRAN/GPU"
-  options(repos = cran)
-  install.packages("mxnet")
+```sh
+git clone --recursive https://github.com/dmlc/mxnet
```
Comment (lupesko) [inline docs/install/windows_setup.md]: "https://github.com/dmlc/mxnet" is out of date. Please use "https://github.com/apache/incubator-mxnet"

File: docs/install/windows_setup.md [hunk 3]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 ### Installing MXNet on a Computer with a GPU Processor
+To install MXNet on a computer with a GPU processor, choose from two options:
+
+* Use the prebuilt binary package
+* Build the library from source code
+
+However, few dependencies remains same for both options.  You will need the following:
 
-To install MXNet R package on a computer with a GPU processor, you need the following:
+* Install Microsoft Visual Studio 2017 (required by CUDA)
```
Comment (lupesko) [inline docs/install/windows_setup.md]: Maybe add link? https://visualstudio.microsoft.com/downloads/

File: docs/install/windows_setup.md [hunk 4]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 
-* Microsoft Visual Studio 2013
+* Install  [NVidia CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit)
 
-* The NVidia CUDA Toolkit
+* Download and install [CuDNN 7](https://developer.nvidia.com/cudnn) (to provide a Deep Neural Network library)
 
-* The MXNet package
+Note: A pre-requisite to above softwares is [Nvidia-drivers](http://www.nvidia.com/Download/index.aspx?lang=en-us) which we assume is installed.
```
Comment (lupesko) [inline docs/install/windows_setup.md]: Why not list as a regular dependency and not a Note?

File: docs/install/windows_setup.md [hunk 5]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
+  options(repos = cran)
+  install.packages("mxnet")
+```
+Change X to 80,90,91 or 92 based on your CUDA toolkit version. Currently, MXNet supports these versions of CUDA.
+#### Building MXNet from Source Code(GPU)
+After you have installed above softwares 
+1. Clone the MXNet github repo.
 
 ```sh
 git clone --recursive https://github.com/dmlc/mxnet
```
Comment (lupesko) [inline docs/install/windows_setup.md]: Same comment as above, please update the link to point at the new GitHub repo URL

File: docs/install/windows_setup.md (hunk 4 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: Usually it comes pre-installed with Windows system, however this needs to be an additional step in Windows Server images on ec2. So, added as note

File: docs/install/windows_setup.md (hunk 5 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: will fix that

File: docs/install/windows_setup.md (hunk 3 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: Thanks! will add that

File: docs/install/windows_setup.md [hunk 6]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 ### Installing MXNet on a Computer with a GPU Processor
+To install MXNet on a computer with a GPU processor, choose from two options:
+
+* Use the prebuilt binary package
+* Build the library from source code
+
+However, few dependencies remains same for both options.  You will need the following:
 
-To install MXNet R package on a computer with a GPU processor, you need the following:
+* Install [Microsoft Visual Studio 2017](https://visualstudio.microsoft.com/downloads/) (required by CUDA)
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: Several other MXNet-related instructions ask for 2015 (MKL-DNN for example). I think we should have both here and identify any nuance in the setups

File: docs/install/windows_setup.md [hunk 7]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 
-* Microsoft Visual Studio 2013
+* Install  [NVidia CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit)
 
-* The NVidia CUDA Toolkit
+* Download and install [CuDNN 7](https://developer.nvidia.com/cudnn) (to provide a Deep Neural Network library)
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: Can we specify a minor version, or at least say cuDNN > 7.0?

File: docs/install/windows_setup.md (hunk 4 above)
Comment (aaronmarkham) [reply docs/install/windows_setup.md]: Again it might be very helpful to identify a version number. I've had issues with this before and it took a lot of digging to figure out that I needed a driver upgrade.

File: docs/install/windows_setup.md [hunk 8]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 
-* The MXNet package
+Note: A pre-requisite to above softwares is [Nvidia-drivers](http://www.nvidia.com/Download/index.aspx?lang=en-us) which we assume is installed.
 
-* CuDNN (to provide a Deep Neural Network library)
+#### Installing MXNet with the Prebuilt Binary Package(GPU)
+For Windows users, MXNet provides prebuilt binary packages.
+You can install the package directly in the R console after you have the above softwares installed
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: software (it's both singular and plural)

Comment (aaronmarkham) [reply docs/install/windows_setup.md]: Also add a period at the end.

File: docs/install/windows_setup.md [hunk 9]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
 
-1. Install the [CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit). The CUDA Toolkit depends on Visual Studio. To check whether your GPU is compatible with the CUDA Toolkit and for information on installing it, see NVidia's [CUDA Installation Guide](http://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/).
-3. Clone the MXNet github repo.
+```r
+  cran <- getOption("repos")
+  cran["dmlc"] <- "https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/R/CRAN/GPU/cuX"
+  options(repos = cran)
+  install.packages("mxnet")
+```
+Change X to 80,90,91 or 92 based on your CUDA toolkit version. Currently, MXNet supports these versions of CUDA.
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: I believe there's some issue with 9.1 and it is not recommended.

Comment (aaronmarkham) [reply docs/install/windows_setup.md]: How does one check their current version #? If you are installing everything for the first time, maybe the recommendation should be made for the user to install 9.2?

File: docs/install/windows_setup.md [hunk 10]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
+#### Building MXNet from Source Code(GPU)
+After you have installed above softwares
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: software, continue with the following steps to build MXNet:

File: docs/install/windows_setup.md [hunk 11]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
-4. Download prebuilt GPU-enabled MXNet libraries for Windows from https://github.com/yajiedesign/mxnet/releases. You will need `mxnet_x64_vc14_gpu.7z` and `prebuildbase_win10_x64_vc14.7z`.
-5. Download and install [CuDNN](https://developer.nvidia.com/cudnn).
-6. Create a folder called ```R-package/inst/libs/x64```. MXNet supports only 64-bit operating systems, so you need the x64 folder.
-7. Copy the following shared libraries (.dll files) into the ```R-package/inst/libs/x64``` folder:
-```
-cublas64_80.dll
-cudart64_80.dll
-cudnn64_5.dll
-curand64_80.dll
+2. Download prebuilt GPU-enabled MXNet libraries for Windows from https://github.com/yajiedesign/mxnet/releases. You will need `mxnet_x64_vc14_gpu_cuX.7z` and `prebuildbase_win10_x64_vc14.7z` where X stands for your CUDA toolkit version
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: This seems odd. Aren't these are binaries for VS2015 (vc14), but the prerequisite was for VS2017? Is there a binary tagged with vc15 available?

File: docs/install/windows_setup.md [hunk 12]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
     ├── dmlc
     ├── mxnet
-    └── nnvm
+    ├── mshadow
+    └── nnvm 
+    
 ```
-9. Make sure that R is added to your ```PATH``` in the environment variables. Running the ```where R``` command at the command prompt should return the location.
-10. Now open the Windows CMD and change the directory to the `mxnet` folder. Then use the following commands
+6. Make sure that R is added to your ```PATH``` in the environment variables. Running the ```where R``` command at the command prompt should return the location.
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: Make sure that the R executable is....

File: docs/install/windows_setup.md [hunk 13]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
+7. Also make sure that Rtools in installed and added to your ```PATH``` in the environment variables.
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: Rtools executable (or folder path if there's more than one in there...)

File: docs/install/windows_setup.md [hunk 14]
```diff
@@ -114,81 +114,138 @@ For CPU-only package:
...
+8. Temporary patch - im2rec currently results in crashes during the build. Remove the im2rec.h and im2rec.cc files in R-package/src/ from cloned repository and comment out the two im2rec lines in R-package/src/mxnet.cc.
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: You might want to show the two lines. These seems kind of vague and easy to mess up.

File: docs/install/windows_setup.md (hunk 9 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: Suggested 9.2 for first time installation

File: docs/install/windows_setup.md (hunk 11 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: added that both 2015 and 2017 are supported

File: docs/install/windows_setup.md (hunk 6 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: fixed

File: docs/install/windows_setup.md (hunk 7 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: suggested to use latest version

File: docs/install/windows_setup.md (hunk 8 above)
Comment (ankkhedia) [reply docs/install/windows_setup.md]: Thanks for pointing out. Taken care

File: docs/install/windows_setup.md [hunk 15]
```diff
@@ -114,81 +114,166 @@ For CPU-only package:
   install.packages("mxnet")
 ```
 
-For GPU-enabled package:
+#### Building MXNet from Source Code(CPU)
+1. Clone the MXNet github repo.
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: View the markdown output. This numbering is wrapping instead of using new lines.

File: docs/install/windows_setup.md [hunk 16]
```diff
@@ -114,81 +114,166 @@ For CPU-only package:
...
+
 ### Installing MXNet on a Computer with a GPU Processor
+To install MXNet on a computer with a GPU processor, choose from two options:
 
-To install MXNet R package on a computer with a GPU processor, you need the following:
+* Use the prebuilt binary package
+* Build the library from source code
 
-* Microsoft Visual Studio 2013
+However, few dependencies remains same for both options.  You will need the following:
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: a few dependencies remain for both options.

File: docs/install/windows_setup.md [hunk 17]
```diff
@@ -114,81 +114,166 @@ For CPU-only package:
...
-4. Download prebuilt GPU-enabled MXNet libraries for Windows from https://github.com/yajiedesign/mxnet/releases. You will need `mxnet_x64_vc14_gpu.7z` and `prebuildbase_win10_x64_vc14.7z`.
-5. Download and install [CuDNN](https://developer.nvidia.com/cudnn).
-6. Create a folder called ```R-package/inst/libs/x64```. MXNet supports only 64-bit operating systems, so you need the x64 folder.
-7. Copy the following shared libraries (.dll files) into the ```R-package/inst/libs/x64``` folder:
-```
-cublas64_80.dll
-cudart64_80.dll
-cudnn64_5.dll
-curand64_80.dll
+2. Download prebuilt GPU-enabled MXNet libraries for Windows from https://github.com/yajiedesign/mxnet/releases. You will need `mxnet_x64_vc14_gpu_cuX.7z` and `prebuildbase_win10_x64_vc14.7z` where X stands for your CUDA toolkit version
```
Comment (aaronmarkham) [inline docs/install/windows_setup.md]: The numbering is wrapping here too.

**Comment by ankkhedia** [comment]
@aaronmarkham I have addressed all of the comments. Could you take a look?
